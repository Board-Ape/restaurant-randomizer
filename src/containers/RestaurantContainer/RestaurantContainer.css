.card-container-container {
  background-color: black; }

h2 {
  display: flex;
  justify-content: space-around;
  background-color: black;
  color: #460C7F;
  font-size: 40px;
  margin: 0;
  padding-bottom: 20px; }
