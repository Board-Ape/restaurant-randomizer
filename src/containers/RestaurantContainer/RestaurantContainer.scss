.card-container-container {
  max-height: 100vh;
  overflow: scroll;
}

.location-title {
  color: white;
  padding-top: 0px;
  font-size: 22px;
}

.lat-long {
  display: flex;
  color: white;
  width: 100%;
  justify-content: space-evenly;
  background: #313131;
  padding: 7px;
  font-weight: 200;
}
