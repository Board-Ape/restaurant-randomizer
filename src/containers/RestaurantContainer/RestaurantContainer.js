import React, { Component } from 'react';
import { getRestaurants, postLocation } from '../../actions';
import RestaurantCardContainer from '../RestaurantCardContainer/RestaurantCardContainer';
import { connect } from 'react-redux';
import './RestaurantContainer.css';
import PropTypes from 'prop-types';
import shuffle from 'lodash/shuffle';

export class RestaurantContainer extends Component {

  componentDidMount = async () => {
    const currentLocation = await postLocation();
    const nearbyRestaurants = await getRestaurants();
    this.props.storeRestaurants(nearbyRestaurants);
    this.props.storeLocation(currentLocation);
  }

  renderRestaurantCardContainer = () => {
    const restaurantsArray = shuffle(this.props.restaurantNames);
    const restaurantsCardsArray = restaurantsArray.splice(0, 3).map((restaurants, index) => {
      return (
        <RestaurantCardContainer
          key={index}
          restaurants={restaurants}
        />
      );
    });
    return restaurantsCardsArray;
  }

  latitude = () => {
    return this.props.currentLocation.map(coords => coords.lat.toFixed(4));
  }

  longitude = () => {
    return this.props.currentLocation.map(coords => coords.lng.toFixed(4));
  }

  render() {
    return (
      <div className='card-container-container'>
        <div className='lat-long'>
          <div className='location-title'>Location</div>
          <div className='location-title'>Latitude: <span>{this.latitude()}</span></div>
          <div className='location-title'>Longitude: <span>{this.longitude()}</span></div>
        </div>
        <div>{this.renderRestaurantCardContainer()}</div>
      </div>
    );
  }
}

export const mapStateToProps = (store) => {
  return {
    restaurantNames: store.restaurantNames,
    currentLocation: store.currentLocation
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    storeRestaurants: (restaurants) => dispatch(getRestaurants(restaurants)),
    storeLocation: (location) => dispatch(postLocation(location))
  };
};

RestaurantContainer.propTypes = {
  storeRestaurants: PropTypes.func,
  storeLocation: PropTypes.func,
  restaurantNames: PropTypes.array,
  currentLocation: PropTypes.array,
  title: PropTypes.string
};

export default connect(mapStateToProps, mapDispatchToProps)(RestaurantContainer);
