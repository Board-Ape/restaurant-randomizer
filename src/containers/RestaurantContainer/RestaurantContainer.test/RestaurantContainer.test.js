import React from 'react';
import {RestaurantContainer, mapStateToProps, mapDispatchToProps } from '../RestaurantContainer';
import { shallow } from 'enzyme';

describe('RestaurantContainer tests', () => {
  let restaurantContainer;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      restaurantNames: [],
      currentLocation: [],
      storeRestaurants: jest.fn(),
      storeLocation: jest.fn()
    };
    restaurantContainer = shallow(
      <RestaurantContainer {...mockProps}/>
    );
  });

  it('should render correctly', () => {
    expect(restaurantContainer).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(restaurantContainer).toMatchSnapshot();
  });

  it('should mount with the correct elements', () => {
    const expected = ['hello', 'there', 'okay', 'maybe', 'good', 'lodash'].length;

    expect(restaurantContainer.find('div').length).toEqual(expected);
  });
});

describe('mapStateToProps tests', () => {
  let mockStore;
  let results;

  beforeEach(() => {
    mockStore = {
      restaurantNames: ['Giordini', 'Masaki', 'Juniper'],
      currentLocation: ['23.2349823', '12.309123']
    };
    results = mapStateToProps(mockStore);
  });

  it('should pull restaurants data from the store', () => {
    expect(results.restaurantNames).toEqual(mockStore.restaurantNames);
  });

  it('should pull location data from the store', () => {
    expect(results.currentLocation).toEqual(mockStore.currentLocation);
  });
});

describe('mapDispatchToProps tests', () => {
  let mockDispatch;
  let results;

  beforeEach(() => {
    mockDispatch = jest.fn();
    results = mapDispatchToProps(mockDispatch);
  });

  it('should dispatch correctly when storing restaurants', () => {
    results.storeRestaurants();

    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should dispatch correctly when storing location', () => {
    results.storeLocation();

    expect(mockDispatch).toHaveBeenCalled();
  });
});
