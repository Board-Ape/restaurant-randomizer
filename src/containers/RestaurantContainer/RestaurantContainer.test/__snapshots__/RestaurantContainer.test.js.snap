// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantContainer tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantContainer
    currentLocation={Array []}
    restaurantNames={Array []}
    storeLocation={
      [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
      }
    }
    storeRestaurants={
      [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="lat-long"
        >
          <div
            className="location-title"
          >
            Location
          </div>
          <div
            className="location-title"
          >
            Latitude: 
            <span />
          </div>
          <div
            className="location-title"
          >
            Longitude: 
            <span />
          </div>
        </div>,
        <div />,
      ],
      "className": "card-container-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="location-title"
            >
              Location
            </div>,
            <div
              className="location-title"
            >
              Latitude: 
              <span />
            </div>,
            <div
              className="location-title"
            >
              Longitude: 
              <span />
            </div>,
          ],
          "className": "lat-long",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Location",
              "className": "location-title",
            },
            "ref": null,
            "rendered": "Location",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Latitude: ",
                <span />,
              ],
              "className": "location-title",
            },
            "ref": null,
            "rendered": Array [
              "Latitude: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Longitude: ",
                <span />,
              ],
              "className": "location-title",
            },
            "ref": null,
            "rendered": Array [
              "Longitude: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="lat-long"
          >
            <div
              className="location-title"
            >
              Location
            </div>
            <div
              className="location-title"
            >
              Latitude: 
              <span />
            </div>
            <div
              className="location-title"
            >
              Longitude: 
              <span />
            </div>
          </div>,
          <div />,
        ],
        "className": "card-container-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="location-title"
              >
                Location
              </div>,
              <div
                className="location-title"
              >
                Latitude: 
                <span />
              </div>,
              <div
                className="location-title"
              >
                Longitude: 
                <span />
              </div>,
            ],
            "className": "lat-long",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Location",
                "className": "location-title",
              },
              "ref": null,
              "rendered": "Location",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Latitude: ",
                  <span />,
                ],
                "className": "location-title",
              },
              "ref": null,
              "rendered": Array [
                "Latitude: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Longitude: ",
                  <span />,
                ],
                "className": "location-title",
              },
              "ref": null,
              "rendered": Array [
                "Longitude: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
